letsencrypt-auto

```diff
@@ -14,7 +14,19 @@ VENV_PATH=${VENV_PATH:-"$XDG_DATA_HOME/$VENV_NAME"}
 VENV_BIN=${VENV_PATH}/bin
 
 if test "`id -u`" -ne "0" ; then
-  SUDO=sudo
+  if type sudo 1>/dev/null 2>&1; then
+    SUDO=sudo
+  else
+    su_sudo() {
+      args=""
+      while [ $# -ne 0 ]; do
+        args="$args'$(printf "%s" "$1" | sed -e "s/'/'\"'\"'/g")' "
```

Comment:
I'm amazed this works, though it appears to do so across a few different shells and execution cases that I've tested.  It's probably robust enough for this use case (which is just the $SUDO'd commmands in this file), though it might be good to add a comment explaining it a little: why do `'` quotes need special treatment when other special quote types (<tt>` " \ </tt>) don't seem to?